website/docs/r/team.html.markdown

```diff
@@ -0,0 +1,45 @@
+---
+layout: "mongodbatlas"
+page_title: "MongoDB Atlas: team"
+sidebar_current: "docs-mongodbatlas-resource-team"
+description: |-
+    Provides a Team resource.
+---
+
+# mongodbatlas_team
+
+`mongodbatlas_team` provides a Team resource. The resource lets you create, edit and delete teams. The resource requires your Organization ID.
+
+-> **NOTE:** Groups and projects are synonymous terms. You may find group_id in the official documentation.
+
+## Example Usage
+
+```hcl
+resource "mongodbatlas_team" "test" {
+    org_id    = "<Your Organization ID>"
+    name      = "myNewTeam"
+    usernames = ["user1", "user2", "user3"]
+}
+```
+
```

Comment:
I was thinking of creating another attribute like optional to assign the roles for the team. I mean, first, create the team, assign the users, later if exists roles for the team attach them to it (using the update API entry point to do that), it would be more comfortable than has another resource. However, we have a bit problem that doesn't exist a GET method or the way to retrieve the team roles (when you create or get a team by id, the roles don't get, I don't know why if the roles is part of the team), just updating it.

Reply:
Ah, okay I think it will help to explain Teams more.   I'll try here and I think this will show us what is (potentially) missing:
A Team is created at the organization level.  Users are added to the team at the organization level.
But then a Team is given access to a Project (so this is at the project level) and then the Team is given roles for that project.   

So the issue seems that we do not have an API method documented for adding a team to a project or for reading what what roles are assigned (but we can add roles to a team via PATCH), does that sound right?  If so I'll follow up internally to figure this out.

Reply:
I still think the core is missing here is that a team can be created before assigning to a project.  With this resource as is it's not clear how to create a team and assign it to more than one project w/ the provider.  For example if I have a team called Support and I create it for my org I might want to add it to 3 different projects, I don't see how that would happen here?  Would you submit the same org and each project in 3 different resource blocks?  If so then the users in each makes even less sense.  Basically it's 1 team to, possibly, 1 or more projects.  The team, however, would have same users - the roles could be different for each project. Happy to setup time to talk about this if we need to @marinsalinas  and @PacoDw      

And just to cover our bases of what we covered in slack quite a few of the team/project related resources are in projects within our documentation https://docs.atlas.mongodb.com/reference/api/project-add-team/  and https://docs.atlas.mongodb.com/reference/api/project-get-teams/

Reply:
Make sense to me @themantissa